x-pack/plugins/ml/public/application/model_management/test_models/models/text_expansion/text_expansion_input.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC } from 'react';
+import useObservable from 'react-use/lib/useObservable';
+import { i18n } from '@kbn/i18n';
+
+import { EuiSpacer, EuiFieldText, EuiFormRow } from '@elastic/eui';
+
+import { TextInput } from '../text_input';
+import { TextExpansionInference } from './text_expansion_inference';
+import { INPUT_TYPE, RUNNING_STATE } from '../inference_base';
+
+const QueryInput: FC<{
+  inferrer: TextExpansionInference;
+}> = ({ inferrer }) => {
+  const questionText = useObservable(inferrer.getQueryText$(), inferrer.getQueryText());
```

Comment:
⚠️  `inferrer.getQueryText$()` returns a new Observable instance on each call (because `.asObservable()` returns a new instance). It's worth exposing the same instance to avoid a render loop in case of the parent component re-render.

Reply:
Discussed this offline and we've agreed that it isn't causing a problem at the moment.
To apply the pattern suggested would cause a bit of churn in the `InferenceBase` class and all derived classes.